internal/controllers/machineset/machineset_controller.go

```diff
@@ -1611,10 +1628,23 @@ func (r *Reconciler) createInfraMachine(ctx context.Context, ms *clusterv1.Machi
 		return nil, clusterv1.ContractVersionedObjectReference{}, errors.Wrapf(err, "failed to create InfraMachine")
 	}
 
-	if err := r.Client.Create(ctx, infraMachine); err != nil {
+	// Create the full object with capi-kubeadmcontrolplane.
+	// Below ssa.RemoveManagedFieldsForLabelsAndAnnotations will drop ownership for labels and annotations
+	// so that in a subsequent syncMachines call capi-kubeadmcontrolplane-metadata can take ownership for them.
```

Comment:
```suggestion
	// Create the full object with capi-machineset.
	// Below ssa.RemoveManagedFieldsForLabelsAndAnnotations will drop ownership for labels and annotations
	// so that in a subsequent syncMachines call capi-machineset-metadata can take ownership for them.
```